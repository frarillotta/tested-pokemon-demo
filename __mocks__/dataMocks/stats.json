[
	{
	   "base_stat":44,
	   "effort":0,
	   "stat":{
		  "name":"hp",
		  "url":"https://pokeapi.co/api/v2/stat/1/"
	   }
	},
	{
	   "base_stat":48,
	   "effort":0,
	   "stat":{
		  "name":"attack",
		  "url":"https://pokeapi.co/api/v2/stat/2/"
	   }
	},
	{
	   "base_stat":65,
	   "effort":1,
	   "stat":{
		  "name":"defense",
		  "url":"https://pokeapi.co/api/v2/stat/3/"
	   }
	},
	{
	   "base_stat":50,
	   "effort":0,
	   "stat":{
		  "name":"special-attack",
		  "url":"https://pokeapi.co/api/v2/stat/4/"
	   }
	},
	{
	   "base_stat":64,
	   "effort":0,
	   "stat":{
		  "name":"special-defense",
		  "url":"https://pokeapi.co/api/v2/stat/5/"
	   }
	},
	{
	   "base_stat":43,
	   "effort":0,
	   "stat":{
		  "name":"speed",
		  "url":"https://pokeapi.co/api/v2/stat/6/"
	   }
	}
 ]