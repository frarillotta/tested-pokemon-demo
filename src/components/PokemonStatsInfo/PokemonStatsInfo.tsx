import styled from "styled-components";
import { getColorByStat, capitalizeFirstLetter } from "../../utils";
import type {Stat} from "../../types";
import { StatCircle } from "../StatCircle/StatCircle";

const PokemonStatsInfo = ({stats}: {stats: Stat[]}) => {

	return <PokemonStatsWrapper>
		{stats.map((stat) => {
			const statName = stat.stat.name;
			return <div key={statName}>
				<StatsWrapper>
					<StatProperty>
						{capitalizeFirstLetter(statName)}
					</StatProperty>
					<StatValue
						color={getColorByStat(statName)}
						value={stat.base_stat}
					/>
					<EffortProperty>
						Effort
					</EffortProperty>
					<EffortValue>
						{stat.effort}
					</EffortValue>
				</StatsWrapper>
			</div>
		})}
	</PokemonStatsWrapper>
}

const PokemonStatsWrapper = styled.div`
	display: grid;
	gap: 20px;
	padding: 25px;
`



const StatsWrapper = styled.div`
	display: grid;
	grid-template-areas: 
		"name value effort effortValue";
	grid-template-columns: repeat(4, 1fr);
	align-items: center;
	gap: 2rem;
`

const StatProperty = styled.span`
	grid-area: name;
	color: #878787
`

const StatValue = styled(StatCircle)`
	grid-area: value;
`

const EffortProperty = styled.span`
	grid-area: effort;
	color: #878787
`

const EffortValue = styled.div`
	grid-area: effortValue;
`

export { PokemonStatsInfo }