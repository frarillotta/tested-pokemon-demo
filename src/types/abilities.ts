import type { URLString } from './pokemon';

export type Abilities = {
    effect_changes: [],
    effect_entries: {
          effect: string,
          language:{
             name: string,
             url:URLString
          },
          short_effect:string
    }[],
    flavor_text_entries: {
          flavor_text: string,
          language:{
             name: string,
             url: URLString
          },
          version_group:{
            name: string,
            url: URLString
          }
    }[],
    generation:{
       name: string,
       url: URLString
    },
    id: number,
    is_main_series: boolean,
    name: string,
    names:{
          language:{
            name: string,
            url: URLString
          },
          name: string
       }[],
    pokemon: {
          is_hidden: boolean,
          pokemon:{
            name: string,
            url: URLString
          },
          slot: number
    }[]
 }