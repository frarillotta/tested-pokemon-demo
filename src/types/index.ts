export type { pokemonType, pokemonResults } from './pokemonsList'; 
export type { Pokemon, GenerationSprites, URLString, Stat, Move } from "./pokemon"