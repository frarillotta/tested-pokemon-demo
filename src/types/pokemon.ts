import { GenerationSprites } from './pokemon';
export type Pokemon = {
    abilities: {
          ability:{
             name: string,
             url: URLString
          },
          is_hidden: boolean,
          slot: number
       }[],
    base_experience: number,
    forms: {
          name: string,
          url:URLString
       }[],
    game_indices: {
          game_index: number,
          version:{
             name: string,
             url: URLString
          }
    }[],
    height: number,
    held_items: any[],
    id: number,
    is_default: boolean,
    location_area_encounters: string,
    moves: Move[],
    name: string,
    order: number,
    past_types: [],
    species:{
       name: string,
       url: string
    },
    sprites:{
		back_default: URLString | null,
		back_female: URLString | null,
		back_shiny: URLString | null,
		back_shiny_female: URLString | null,
		front_default: URLString | null,
		front_female: URLString | null,
		front_shiny: URLString | null,
		front_shiny_female: URLString | null,
       other:{
          dream_world: GenerationSprites,
          "official-artwork": GenerationSprites
       },
       versions:{
          "generation-i": {
             "red-blue": GenerationSprites,
             yellow: GenerationSprites
          },
            "generation-ii": {
             crystal: GenerationSprites,
             gold: GenerationSprites,
             silver: GenerationSprites
          },
          "generation-iii":{
             emerald: GenerationSprites,
             "firered-leafgreen": GenerationSprites,
             "ruby-sapphire": GenerationSprites
          },
          "generation-iv":{
             "diamond-pearl": GenerationSprites,
             "heartgold-soulsilver": GenerationSprites,
             platinum: GenerationSprites
          },
          "generation-v":{
             "black-white":{
                animated: GenerationSprites,
                GenerationSprites
             }
          },
          "generation-vi":{
             "omegaruby-alphasapphire": GenerationSprites,
             "x-y": GenerationSprites
          },
          "generation-vii":{
             icons: GenerationSprites,
             "ultra-sun-ultra-moon": GenerationSprites
          },
          "generation-viii":{
             icons: GenerationSprites
          }
       }
    },
    stats: Stat[],
    types: {
          slot: number,
          type:{
             name: "normal" | "fire" | "water" | "grass" | "electric" | "ice" | "fighting" | "poison" | "ground" | "flying" | "psychic" | "bug" | "rock" | "ghost" | "dark" | "dragon" | "steel" | "fairy",
             url: URLString
          }
       }[],
    weight: number
 }

export type GenerationSprites = {
    back_default: URLString | null,
    back_female: URLString | null,
    back_shiny: URLString | null,
    back_shiny_female: URLString | null,
    front_default: URLString | null,
    front_female: URLString | null,
    front_shiny: URLString | null,
    front_shiny_female: URLString | null
 };

export type Stat = {
	base_stat: number,
	effort: number,
	stat:{
	   name: "hp" | "attack" | "defense" | "special-defense" | "special-attack" | "speed",
	   url: URLString
	}
};

export type Move = {
	move:{
	   name: string,
	   url: URLString
	},
	version_group_details: {
		  level_learned_at: number,
		  move_learn_method:{
			 name: string,
			 url: URLString
		  },
		  version_group:{
			 name: string,
			 url: URLString
		  }
	  }[],
 }

export type URLString = `https://${string}`;
