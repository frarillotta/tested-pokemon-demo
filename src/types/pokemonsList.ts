export type pokemonType = {
    count: number,
    next: string,
    previous: string,
    results: pokemonResults[]
}

export type pokemonResults = {
    name: string, 
    url: string
}