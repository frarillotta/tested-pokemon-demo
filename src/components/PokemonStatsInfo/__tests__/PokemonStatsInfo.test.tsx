import { render } from "@testing-library/react";
import {PokemonStatsInfo} from "../PokemonStatsInfo";
import stats from "../../../../__mocks__/dataMocks/stats.json";
import { Stat } from "../../../types";

describe("PokemonStatsInfo", () => {
	
	describe("when loaded", () => {
		
		it("should have a consistent snapshot", () => {
			const {container} = render(<PokemonStatsInfo 
				stats={stats as Stat[]}
			/>);
			expect(container).toMatchSnapshot();
		});

	});

})