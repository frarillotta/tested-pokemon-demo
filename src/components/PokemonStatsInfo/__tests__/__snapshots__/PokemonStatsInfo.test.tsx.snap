// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeballButton when isCaptured=true should have a consistent snapshot 1`] = `
<div>
  <div
    class="PokemonStatsInfo__PokemonStatsWrapper-sc-16epyc7-0 dOyVIv"
  >
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Hp
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(204, 68, 52);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Attack
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(240, 128, 48);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Defense
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(248, 208, 48);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          1
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Special-attack
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(104, 144, 240);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Special-defense
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(84, 166, 102);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Speed
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(252, 104, 152);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PokemonStatsInfo when loaded should have a consistent snapshot 1`] = `
<div>
  <div
    class="PokemonStatsInfo__PokemonStatsWrapper-sc-16epyc7-0 dOyVIv"
  >
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Hp
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(204, 68, 52);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Attack
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(240, 128, 48);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Defense
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(248, 208, 48);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          1
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Special-attack
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(104, 144, 240);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Special-defense
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(84, 166, 102);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
    <div>
      <div
        class="PokemonStatsInfo__StatsWrapper-sc-16epyc7-1 ePiRYv"
      >
        <span
          class="PokemonStatsInfo__StatProperty-sc-16epyc7-2 jpsEst"
        >
          Speed
        </span>
        <div
          class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu PokemonStatsInfo__StatValue-sc-16epyc7-3 bUZoGr"
        >
          <div
            aria-valuenow="0"
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
            role="progressbar"
            style="width: 50px; height: 50px; transform: rotate(-90deg); color: rgb(252, 104, 152);"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
                style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
              />
            </svg>
          </div>
          <div
            class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
          >
            <div
              class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
            >
              0
            </div>
          </div>
        </div>
        <span
          class="PokemonStatsInfo__EffortProperty-sc-16epyc7-4 iYkOYG"
        >
          Effort
        </span>
        <div
          class="PokemonStatsInfo__EffortValue-sc-16epyc7-5 gfPJIT"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
